03week/towersOfHanoi.js

```diff
@@ -13,61 +13,191 @@ let stacks = {
   c: []
 };
 
+//Print the currenct stack elements
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+//This method validates weather the source and destination stacks are
+//valid. It can be only a,b or c.
+function validateStackEntries(startStack, endStack){
+ if((startStack === 'a') || (startStack === 'b') || (startStack === 'c')) 
+ {
+   if(startStack === endStack)
+     return false;
+    if((endStack === 'a') || (endStack === 'b') || (endStack === 'c'))
+     return true;
+   else
+     return false;
+ }
+ else
+   return false;  
 }
+//This function will perform the mobe from source stack to destination stack
+function movePiece(startStack,endStack) {
+  if((startStack === 'a') && (endStack === 'b'))
```

Comment:
for every stack and letter, you're checking if the stack is a,b, or c, twice, that's a lot of repetition that you can condense.